src/ui.rs

```diff
@@ -40,6 +39,16 @@ pub enum Theme {
     Funky,
 }
 
+/// The draw mode applied to a group of objects in the viewport. Not always a
+/// 1:1 mapping with renderer materials.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum ViewportDrawMode {
```

Comment:
This doesn't need to be `Eq` or `Hash`.

Reply:
Hmmm, I can remove `Hash`, but I usually prefer to keep `Eq` to state future intentions, that this enum will always be `Eq`.